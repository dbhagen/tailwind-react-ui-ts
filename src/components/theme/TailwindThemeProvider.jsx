import React from 'react'
import PropTypes from 'prop-types'
import merge from 'lodash.merge'

import defaultTheme from './defaultTheme'
import TailwindTheme from './TailwindTheme'

const TailwindThemeProvider = ({ theme, children }) => {
  const mergedTheme = merge({}, defaultTheme, theme)

  return (
    <TailwindTheme.Provider value={mergedTheme}>
      {children}
    </TailwindTheme.Provider>
  )
}

TailwindThemeProvider.propTypes = {
  theme: PropTypes.shape({}),
  children: PropTypes.node,
}

TailwindThemeProvider.defaultProps = {
  theme: {},
  children: undefined,
}

export default TailwindThemeProvider
