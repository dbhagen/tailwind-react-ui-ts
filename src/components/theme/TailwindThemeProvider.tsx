import React from 'react'
import merge from 'lodash.merge'

import defaultTheme from './defaultTheme'
import TailwindTheme from './TailwindTheme'

export type TailwindThemeProviderProps = {
  theme: {}
  children: React.ReactNode
}

const TailwindThemeProvider: React.FunctionComponent<TailwindThemeProviderProps> = ({ theme, children }) => {
  const mergedTheme = merge({}, defaultTheme, theme)

  return (
    <TailwindTheme.Provider value={mergedTheme}>
      {children}
    </TailwindTheme.Provider>
  )
}

export default TailwindThemeProvider
