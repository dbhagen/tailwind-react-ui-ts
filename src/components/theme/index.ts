export { default as defaultTheme } from './defaultTheme'
export { default as TailwindTheme } from './TailwindTheme'
export { default as TailwindThemeProvider } from './TailwindThemeProvider'
export { default as withTheme } from './withTheme'
export { default as useThemeValue } from './useThemeValue'
