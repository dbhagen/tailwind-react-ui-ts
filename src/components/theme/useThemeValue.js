import { tailwindPropToClassName } from '../tailwind'

export default (prefix, value, userClassNames = '') => {
  if (userClassNames.includes(prefix)) {
    return false
  }

  return tailwindPropToClassName(prefix, value)
}
