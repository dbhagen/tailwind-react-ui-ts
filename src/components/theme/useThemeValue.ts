import { tailwindPropToClassName } from '../tailwind'

export default (prefix: string, values: Object, userClassNames = '') => {
  if (userClassNames.includes(prefix)) {
    return false
  }

  return tailwindPropToClassName(undefined, values, prefix)
}
