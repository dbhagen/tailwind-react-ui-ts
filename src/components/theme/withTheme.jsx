import React from 'react'

import TailwindTheme from './TailwindTheme'

const withTheme = Component => {
  const WithTheme = props => (
    <TailwindTheme.Consumer>
      {theme => <Component {...props} theme={theme} />}
    </TailwindTheme.Consumer>
  )

  WithTheme.displayName = `WithTheme(${Component.displayName})`

  return WithTheme
}

export default withTheme
