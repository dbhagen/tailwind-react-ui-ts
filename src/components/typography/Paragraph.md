```jsx
<Paragraph lead>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
  incididunt ut labore et dolore magna aliqua.
</Paragraph>

<Paragraph>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
  incididunt ut labore et dolore magna aliqua.
</Paragraph>
```
