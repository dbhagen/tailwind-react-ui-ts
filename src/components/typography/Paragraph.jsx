import React from 'react'
import PropTypes from 'prop-types'

import { withTheme } from '../theme'
import { Text } from '../primitives'
import { propTypes } from '../tailwind'
import { getAsArray } from '../utils'

const Paragraph = ({
  theme,
  children,
  is,
  size,
  lead,
  brand,
  paragraph,
  text,
  ...rest
}) => (
  <Text
    is={is}
    text={[
      (size || lead) &&
        theme.text.size.body[(lead ? theme.text.size.body.length : size) - 1],
      ...getAsArray(text),
    ]}
    m={{ b: theme.spacing.md }}
    {...rest}
  >
    {children}
  </Text>
)

Paragraph.propTypes = {
  theme: PropTypes.shape({}).isRequired,
  children: PropTypes.node,
  is: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
  size: PropTypes.number,
  lead: PropTypes.bool,
  brand: PropTypes.bool,
  paragraph: PropTypes.bool,
  text: propTypes.text,
}

Paragraph.defaultProps = {
  children: undefined,
  is: 'p',
  size: undefined,
  paragraph: false,
  lead: false,
  brand: false,
  text: undefined,
}

export { Paragraph as component }
export default withTheme(Paragraph)
