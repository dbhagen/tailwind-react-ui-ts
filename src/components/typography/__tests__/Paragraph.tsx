import React from 'react'
import { shallow } from 'enzyme'

import { component as Paragraph } from '../Paragraph'
import { defaultTheme } from '../../theme'

const setup = (testProps = {}) => {
  const props = Object.assign({ theme: defaultTheme }, testProps)

  const wrapper = shallow(
    <Paragraph {...props}>Lorem ipsum dolor sit amet</Paragraph>,
  )

  return {
    props,
    wrapper,
  }
}

describe('Paragraph', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup({ is: 'p' })

    expect(wrapper).toMatchSnapshot()
  })
})
