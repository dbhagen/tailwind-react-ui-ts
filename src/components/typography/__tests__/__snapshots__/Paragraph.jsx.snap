// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders matching snapshot 1`] = `
<Text
  bold={false}
  is="p"
  leading="normal"
  loose={false}
  m={
    Object {
      "b": 4,
    }
  }
  text={
    Array [
      false,
    ]
  }
  tight={false}
>
  Lorem ipsum dolor sit amet
</Text>
`;
